### PR TITLE
Migrate to CircleCI, Rubcop updates, shift test DB to use SQLite

### PR DESCRIPTION
**Pull request by skarger**

### CODE REVIEW

File: spec/delayed_job_chainable_hooks_spec.rb [hunk 1]
```diff
@@ -78,127 +78,5 @@
         expect { Delayed::Worker.new.work_off(1) }.not_to raise_error
       end
     end
-
```
Comment (skarger) [inline spec/delayed_job_chainable_hooks_spec.rb:81]: Rubocop doesn't want us to dynamically define constants in specs. However, using `let ... { Class.new }` does not work for these tests, due to subtle difference with class variables and inheritance. So I moved these test classes into `spec/fixtures`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
-## v1.0.0
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+
+## Unreleased
+
+## 2.0.0 - 2021-03-31
+### Changed
+- Require Ruby 2.7+
```
Comment (jturkel) [inline CHANGELOG.md]: Most of our libraries require Ruby 2.6+ since Ruby 2.6 won't be EOL until March 31, 2022. How painful is support Ruby 2.6?

File: config.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+version: 2.1
```
Comment (jturkel) [inline config.yml]: Did you mean to check this file in since the CircleCI config already exists in `.circleci/config.yml`?

File: spec/delayed_job_chainable_hooks_spec.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
+# frozen_string_literal: true
+
 describe DelayedJobChainableHooks do
   before do
+    spec_dir = File.dirname(__FILE__)
+    require "#{spec_dir}/fixtures/job_classes"
```
Comment (jturkel) [inline spec/delayed_job_chainable_hooks_spec.rb]: Should we require the file from `spec_helper` so we get more consistent constant loading across tests? At a minimum this doesn't need to be done in a `before` block and could utilize `require_relative`.

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -28,29 +31,44 @@
   # you configure your source control system to ignore this file.
   config.example_status_persistence_file_path = 'spec/examples.txt'
 
-  DATABASE_NAME = 'delayed_job_chainable_hooks_test'.freeze
```
Comment (jturkel) [inline spec/spec_helper.rb:31]: It doesn't look like this gem actually depends on Postgres. Should we switch to sqlite for simpler test setup? [goldiloader](https://github.com/salsify/goldiloader) uses it for tests.

File: CHANGELOG.md (hunk 2 above)
Comment (skarger) [reply CHANGELOG.md]: It should be straightforward to include.

File: config.yml (hunk 3 above)
Comment (skarger) [reply config.yml]: Just an oversight, I'd started this before a meeting and then came back to it.

File: spec/delayed_job_chainable_hooks_spec.rb (hunk 4 above)
Comment (skarger) [reply spec/delayed_job_chainable_hooks_spec.rb]: Sure. I'd put it in spec_helper.rb first and then moved it here for context, but I was on the fence.

File: spec/spec_helper.rb (hunk 5 above)
Comment (skarger) [reply spec/spec_helper.rb:31]: After I had so much fun configuring Postgres for CircleCI...

I'll give sqlite a try, makes sense to simplify if it's not too hard to set up.

**Comment by skarger** [comment]
@jturkel addressed your comments.

File: CHANGELOG.md (hunk 2 above)
Comment (jturkel) [inline CHANGELOG.md]: This should be updated to reflect Ruby 2.6 support
```suggestion
- Require Ruby 2.6+
```

File: delayed_job_chainable_hooks.gemspec [hunk 6]
```diff
@@ -29,19 +30,22 @@ Gem::Specification.new do |spec|
...
-  spec.add_development_dependency 'rspec', '~> 3.4'
-  spec.add_development_dependency 'simplecov'
-  spec.add_development_dependency 'salsify_rubocop', '~> 0.48.1'
-  spec.add_development_dependency 'overcommit'
-  spec.add_development_dependency 'activerecord', '>= 4.2'
+  spec.add_development_dependency 'activerecord', '>= 5.2'
+  spec.add_development_dependency 'bundler', '~> 2.0'
   spec.add_development_dependency 'database_cleaner'
+  spec.add_development_dependency 'overcommit'
   spec.add_development_dependency 'pg'
```
Comment (jturkel) [inline delayed_job_chainable_hooks.gemspec]: We can remove the `pg` dependency now that we're using sqlite.

**Review by jturkel — APPROVED** [review]
